src/main/java/space/obminyashka/items_exchange/rest/LocationController.java

```diff
@@ -72,6 +72,21 @@ public List<LocationNameView> getAllDistricts(@RequestParam UUID areaId) throws
         return locationNameViewList;
     }
 
+    @GetMapping(value = ApiKey.LOCATION_CITY, produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get all city by district id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    @ResponseStatus(HttpStatus.OK)
+    public List<LocationNameView> getAllCityByDistrictId(@RequestParam UUID districtId) throws EntityIdNotFoundException{
+        List<LocationNameView> locationNameViewList = locationService.getAllCityByDistrictId(districtId);
+        if (locationNameViewList.isEmpty()) {
+            throw new EntityIdNotFoundException(getParametrizedMessageSource(INVALID_LOCATION_ID, districtId));
+        }
```

Comment:
It'd be way better if you check the existence of the District ID before fetching the results.

Reply:
done